dev/com.ibm.ws.beanvalidation.v20.cdi/src/com/ibm/ws/beanvalidation/v20/cdi/internal/LibertyHibernateValidatorExtension.java

```diff
@@ -194,6 +194,20 @@ private ClassLoadingService getClassLoadingService() {
         return classLoadingService;
     }
 
+    private String getAppName() {
+        // Get the CDIService
+        Bundle bundle = FrameworkUtil.getBundle(CDIService.class);
+        CDIService cdiService = AccessController.doPrivileged((PrivilegedAction<CDIService>) () -> {
+            BundleContext bCtx = bundle.getBundleContext();
+            ServiceReference<CDIService> svcRef = bCtx.getServiceReference(CDIService.class);
+            return svcRef == null ? null : bCtx.getService(svcRef);
+        });
+        if (cdiService == null) {
+            return null;
+        }
```

Comment:
a) why get hold of CDIService the "hard" way? Why not just inject a Reference?
b) you're calling this method from within a CDI Extension method ... if the CDISerivce is null then there's something very odd happening!

Reply:
for (a) I didn't want to define a new service dependency, since this reference will only ever be used on an error path (most of the time it will never be needed at all).  
for (b) the extra null check was just my tinfoil hat here... wasn't sure what sort of things might be going wrong if we're having problems registering the CDI extension so I didn't want to make any extra assumptions.
I can resolve (b), but would prefer to keep (a) the way it is